project.janet

```diff
@@ -5,7 +5,8 @@
   :dependencies ["https://github.com/janet-lang/spork.git"
                  "https://github.com/CFiggers/jayson.git"
                  "https://github.com/ianthehenry/judge.git"
-                 "https://github.com/CFiggers/cmd.git"])
+                 "https://github.com/CFiggers/cmd.git"
+                 "https://github.com/CFiggers/jgraph.git"])
```

Comment:
jgraph isn't really a dependency of this project, it's just referenced in some sample code used to test the formatter. I wouldn't include it here (if having it as part of the formatting example is causing problems, I'd rather get a new formatting sample).

Reply:
FWIW, I had noticed that without `jgraph` installed the tests (at least those invokable via `jpm test`) don't run without errors:

```
$ jpm test
running test/test-format-file-after.janet ...
error: could not find module jgraph:
    build/jgraph.so
    /home/user/.local/lib/janet/jgraph.jimage
    /home/user/.local/lib/janet/jgraph.janet
    /home/user/.local/lib/janet/jgraph/init.janet
    /home/user/.local/lib/janet/jgraph.so
  in require-1 [boot.janet] on line 3028, column 20
  in import* [boot.janet] on line 3072, column 15
  in thunk [test/test-format-file-after.janet] (tail call) on line 3, column 1
non-zero exit code in test/test-format-file-after.janet: 1
running test/test-format-file-before.janet ...
error: could not find module jgraph:
    build/jgraph.so
    /home/user/.local/lib/janet/jgraph.jimage
    /home/user/.local/lib/janet/jgraph.janet
    /home/user/.local/lib/janet/jgraph/init.janet
    /home/user/.local/lib/janet/jgraph.so
  in require-1 [boot.janet] on line 3028, column 20
  in import* [boot.janet] on line 3072, column 15
  in thunk [test/test-format-file-before.janet] (tail call) on line 3, column 1
non-zero exit code in test/test-format-file-before.janet: 1
running test/test-integration.janet ...
running test/test-lookup.janet ...
running test/test-main.janet ...
Starting LSP
```

Reply:
Yeah, I don't use `jpm test` in this project—I'm using [ianthehenry/judge](https://github.com/ianthehenry/judge) for everything (admittedly not all of those are passing at the moment either... 😅)

Reply:
The tests *were* running with `jpm test` when I ran them (with jgraph installed). I didn't realize judge had its own executable.

Reply:
Tests set up with Judge (`(test)` and `(deftest)` macros) don't mean anything when run with `jpm test`—`jpm test` essentially just evaluates `.janet` files, it doesn't do any special test running. The `(test)` and `(deftest)` macros in Judge evaluate to no-ops when you're not using the `judge` binscript. So even failing Judge tests will not complain when `jpm test` is run.

Reply:
FWIW, I disabled `jpm test` in a project like [this](https://github.com/sogaiu/tree-sitter-janet-simple/blob/6bfbaadac2ba0da21087041eff85d26129c4c920/project.janet#L43-L46).

Reply:
Could we use jpm to invoke judge instead? I usually a pretty big fan of consistent interfaces :)

Reply:
[`judge`'s documentation](https://github.com/ianthehenry/judge/#usage) mentions a specific way it can be used via `project.janet`:

> You can also add this to your project.janet file:

> (task "test" [] (shell "jpm_tree/bin/judge"))

> To run Judge with a normal jpm test invocation. 

May be that's suitable for janet-lsp's case?

Reply:
> [`judge`'s documentation](https://github.com/ianthehenry/judge/#usage) mentions a specific way it can be used via `project.janet`:
> 
> > You can also add this to your project.janet file:
> 
> > (task "test" [] (shell "jpm_tree/bin/judge"))
> 
> > To run Judge with a normal jpm test invocation. 
> 
> May be that or something similar is suitable for janet-lsp's case?

I'm trying to get this working and right now I'm only getting obscure build errors whenever this task is included in `project.janet`.